tests/all.cs

```diff
@@ -876,6 +876,191 @@ public void TestAdd()
     CommonChecks(intp);
   }
 
+  [IsTested()]
+  public void TestPostOpAddAssign()
+  {
+    string bhl = @"
+      
+    func int test() 
+    {
+      int k
+      k += 10
+      return k
+    }
+    ";
+
+    var intp = Interpret(bhl);
+    var node = intp.GetFuncCallNode("test");
+    var num = ExtractNum(ExecNode(node));
+
+    AssertEqual(num, 10);
+    CommonChecks(intp);
+  }
+
+  [IsTested()]
+  public void TestPostOpSubAssign()
+  {
+    string bhl = @"
+      
+    func int test() 
+    {
+      int k
+      k -= 10
+      return k
+    }
+    ";
+
+    var intp = Interpret(bhl);
+    var node = intp.GetFuncCallNode("test");
+    var num = ExtractNum(ExecNode(node));
+
+    AssertEqual(num, -10);
+    CommonChecks(intp);
+  }
+
+  [IsTested()]
+  public void TestPostOpMulAssign()
+  {
+    string bhl = @"
+      
+    func int test() 
+    {
+      int k = 1
+      k *= 10
+      return k
+    }
+    ";
+
+    var intp = Interpret(bhl);
+    var node = intp.GetFuncCallNode("test");
+    var num = ExtractNum(ExecNode(node));
+
+    AssertEqual(num, 10);
+    CommonChecks(intp);
+  }
+
+  [IsTested()]
+  public void TestPostOpDivAssign()
+  {
+    string bhl = @"
+      
+    func int test() 
+    {
+      int k = 10
+      k /= 10
+      return k
+    }
+    ";
+
+    var intp = Interpret(bhl);
+    var node = intp.GetFuncCallNode("test");
+    var num = ExtractNum(ExecNode(node));
+
+    AssertEqual(num, 1);
+    CommonChecks(intp);
+  }
+
+  [IsTested()]
+  public void TestPostOpAddAssignStringNotAllowed()
+  {
+    string bhl = @"
+      
+    func void test() 
+    {
+      string k
+      k += ""foo""
+    }
+    ";
+
+    AssertError<UserError>(
+      delegate() { 
+        Interpret(bhl);
+      },
+      " : PostOpAssign not supported for string"
+    );
+  }
+
+  [IsTested()]
+  public void TestPostOpSubAssignStringNotAllowed()
+  {
+    string bhl = @"
+      
+    func void test() 
+    {
+      string k
+      k -= ""foo""
+    }
+    ";
+
+    AssertError<UserError>(
+      delegate() { 
+        Interpret(bhl);
+      },
+      " : PostOpAssign not supported for string"
+    );
+  }
+
+  [IsTested()]
+  public void TestPostOpMulAssignStringNotAllowed()
+  {
+    string bhl = @"
+      
+    func void test() 
+    {
+      string k
+      k *= ""foo""
+    }
+    ";
+
+    AssertError<UserError>(
+      delegate() { 
+        Interpret(bhl);
+      },
+      " : PostOpAssign not supported for string"
+    );
+  }
+
+  [IsTested()]
+  public void TestPostOpDivAssignStringNotAllowed()
+  {
+    string bhl = @"
+      
+    func void test() 
+    {
+      string k
+      k /= ""foo""
+    }
+    ";
+
+    AssertError<UserError>(
+      delegate() { 
+        Interpret(bhl);
+      },
+      " : PostOpAssign not supported for string"
+    );
+  }
+
+  [IsTested()]
+  public void TestPostOpAssignExpIncompatibleTypesNotAllowed()
```

Comment:
We also need a test for the following case:

float k = 2.1
int a = 1
k += a

In this case types are compatible